samza-core/src/main/scala/org/apache/samza/job/JobRunner.scala

```diff
@@ -118,6 +118,21 @@ class JobRunner(config: Config) extends Logging {
     }
     coordinatorSystemProducer.stop()
 
+
+    // if diagnostics is enabled, create diagnostics stream as specified in config
+    if (new JobConfig(config).getDiagnosticsEnabled) {
+      val systemStreamName = new MetricsConfig(config).
+        getMetricsReporterStream(MetricsConfig.METRICS_SNAPSHOT_REPORTER_NAME_FOR_DIAGNOSTICS).get
+      val systemStream = StreamUtil.getSystemStreamFromNames(systemStreamName)
+      val diagnosticStreamSpec = new StreamSpec("diagnostics-stream-id", systemStream.getStream, systemStream.getSystem)
+      val diagnosticSysAdmin = systemFactory.getAdmin(systemStream.getSystem, config)
+      info("creating stream %s using  sysAdmin %s " format (systemStreamName, diagnosticSysAdmin))
```

Comment:
I don't think SystemAdmin objects in general have a toString, so it might not be meaningful to log it.